### PR TITLE
feat:AddedSearchFunctionality

### PR DESCRIPTION
**Pull request by gavandivya**
image
image

### CODE REVIEW

**Comment by nikitaj-57** [comment]
Reviewed.

File: Lecture4/App.js [hunk 1]
```diff
@@ -1,14 +1,41 @@
 import React from "react";
+import { useState } from "react"
```
Comment (ap221882) [inline Lecture4/App.js]: You can do this-
import React, {useState} from 'react';
Also, a blank line after global import will look more readable.

File: Lecture4/App.js [hunk 2]
```diff
@@ -1,14 +1,41 @@
...
 import { createRoot } from "react-dom/client";
```
Comment (ap221882) [inline Lecture4/App.js:2]: Appreciate it. I have always in my life imported ReactDOM. But you've no use of it here I think.

File: Lecture4/App.js [hunk 3]
```diff
@@ -1,14 +1,41 @@
...
+  return (
+    <>
+      <SearchComponent setFilteredData={setFilteredData} />
+      {filteredData.length != 0 ? <CardContainerComponent filteredData={filteredData} /> : <NoRecord />}
+    </>
+  )
+}
+
+
+const NoRecord = () => (<div style={{ textAlign: "center", color: "white" }}>
```
Comment (ap221882) [inline Lecture4/App.js]: It is good that you're trying inline styling. But don't use it. Not readable at all in large applications.

File: Lecture4/Component/SearchComponent.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { useState } from "react";
+import data from '../data.json'
+
+const searchTeamdetails = (searchedText) => {
```
Comment (ap221882) [inline Lecture4/Component/SearchComponent.js:4]: Try to make it a reusable/utility function.

File: Lecture4/style.css [hunk 5]
```diff
@@ -98,18 +98,32 @@ h6 {
...
-  filter: blur(0) !important;
+}
+
+form {
+  display: flex;
+  justify-content: right;
 }
 
 @media screen and (max-width: 576px) {
   .container {
```
Comment (ap221882) [inline Lecture4/style.css:120]: Appreciated.

File: Lecture4/Component/CardComponent.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
+import data from "../data.json";
+import { SocialMediaIcon as icon } from "../socialMediaIcon";
+import getImageUrl from "simple-image-cdn";
+import React, { useState, useEffect } from "react";
+
+const CardComponent = ({ details }) => {
+  const { profilePic, name, designation, location, github, linkedin, instagram, discord } = details;
```
Comment (ap221882) [inline Lecture4/Component/CardComponent.js:5]: You can destructure it more like 
```
{details: {profilePic, name, designation, location, github, linkedin, instagram, discord}}
```

**Review by ap221882 — COMMENTED** [review]
Very nice work. Added some suggestions.

**Comment by roshantrivedi** [comment]
Nice work.. inline styling can be avoided otherwise everything looks great.

**Pull request merged**